diff --git a/node_modules/openalex-sdk/dist/src/utils/works.js b/node_modules/openalex-sdk/dist/src/utils/works.js
index 49eee4b..a8f18fd 100644
--- a/node_modules/openalex-sdk/dist/src/utils/works.js
+++ b/node_modules/openalex-sdk/dist/src/utils/works.js
@@ -27,8 +27,8 @@ function buildUrl(baseUrl, search, searchField, filter, group_by) {
         filterParams = filterBuilder(filter);
     if (group_by)
         GroupByParams = `group_by=${group_by}`;
-    if (search && searchField)
-        filterParams += `,${searchField}.search:${search}`;
+    if (search && searchField ) filterParams += filter ? `,${searchField}.search:${search}` : `${searchField}.search:${search}`;
+
     if (search && !searchField)
         SearchParams = `search=${search}`;
     if (searchField || filter)
diff --git a/node_modules/openalex-sdk/src/utils/works.ts b/node_modules/openalex-sdk/src/utils/works.ts
index e8f8534..37db168 100644
--- a/node_modules/openalex-sdk/src/utils/works.ts
+++ b/node_modules/openalex-sdk/src/utils/works.ts
@@ -22,7 +22,7 @@ export function buildUrl(baseUrl: string, search?: string, searchField?: string,
   if (filter) filterParams = filterBuilder(filter);
   if (group_by) GroupByParams = `group_by=${group_by}`;
 
-  if (search && searchField) filterParams += `,${searchField}.search:${search}`;
+  if (search && searchField ) filterParams += filter ? `,${searchField}.search:${search}` : `${searchField}.search:${search}`;
   if (search && !searchField) SearchParams = `search=${search}`;
   if (searchField || filter) filterParams = `filter=${filterParams}`;
   return `${baseUrl}/works?${filterParams}&${SearchParams}&${GroupByParams}`;
